GCC: conversion to unsigned int warning ( [-Werror=sign-conversion])

```
third_party_deps/magic_enum-src/include/magic_enum.hpp:733:43: error: conversion to ‘unsigned int’ from ‘int’ may change the sign of the result [-Werror=sign-conversion]
  733 |       crc = (crc >> 8) ^ crc_table[(crc ^ c) & 0xff];
      |                                           ^
```

`c` is signed, but `crc` is unsigned, the conversion is ok, but can you please make it explicit so we don't have to disable this useful GCC warning, locally or globally, just because of this?

```
      crc = (crc >> 8) ^ crc_table[(crc ^ static_cast<std::uint32_t>(c)) & 0xff];
```

--- a/include/magic_enum.hpp
+++ b/include/magic_enum.hpp
@@ -730,7 +730,7 @@ struct constexpr_hash_t<Value, std::enable_if_t<std::is_same_v<Value, string_vie
   constexpr std::uint32_t operator()(string_view val) const noexcept {
     auto crc = static_cast<std::uint32_t>(0xffffffffL);
     for (const auto c : val) {
-      crc = (crc >> 8) ^ crc_table[(crc ^ c) & 0xff];
+      crc = (crc >> 8) ^ crc_table[(crc ^ static_cast<std::uint32_t>(c)) & 0xff];
     }
     return crc ^ 0xffffffffL;
   }

